client/src/components/shared/Info/RecursiveInfo.js

```diff
@@ -0,0 +1,37 @@
+import React, {Component} from 'react'
+import CompanyDetails from '../CompanyDetails'
+import './RecursiveInfo.css'
+
+class RecursiveInfo extends Component {
+  componentWillMount() {
+    const state = {
+      extracted: false,
```

Comment:
This can be converted to functional component using recompose. It will simplify the component a bit